src/log4net.ElasticSearch/Models/Uri.cs

```diff
@@ -2,12 +2,33 @@
 using System.Collections.Specialized;
 using log4net.ElasticSearch.Infrastructure;
 
+#if NETFRAMEWORK
+    using System.Web;
+#else
+    using System.Net;

```

Comment:
Isnt' `System.Net` also available on .NET Framework?
Would it be possible to remove this `#if` and the reference to `System.Web` on the `csproj` and get around just with `System.Net` for the URL encoding?

Reply:
I couldn't get the .NETStandard 2.0 target to build without having a reference to a full framework assembly. While you can reference the full framework assembly and it does compile and work (albeit with a big fat warning), it isn't guaranteed to not blow up at run-time if something in the full framework assembly isn't implemented in .NETStandard.  

I don't like compilation directives either, but when targeting multiple frameworks with the same code base, they're pretty much inevitable.

Reply:
What I meant is to not use the full framework dependency (no `System.Web`) and use the `System.Net` approach for both `net4x` and `netstandard2.0`

Reply:
You could do that if you were targeting NETFramework 4.6.1 or higher, but .NETStarndard 2.0 isn't supported under 4 or 4.5.

https://docs.microsoft.com/en-us/dotnet/standard/net-standard

...and it looks like they re-homed and renamed the URL decoding class.

Reply:
I meant to use WebUtility for both .NET Standard and .NET Framework.

[The docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.webutility?view=netframework-4.8) say it's available since .NET Framework 4.0.

Reply:
Will give it a try...

Reply:
Now I remember test:

When_a_user_or_password_have_special_characters_it_should_still_work()

Expected string to be 
"Basic IUAjPD4sLi9cfCQlPyYqKCk6IUAjJCU9XiYqPSgpX3t9Oj4/Pjw=" with a length of 58, but 
"Basic ISU0MCUyMyUzQyUzRSUyQy4lMkYlNUMlN0MlMjQlMjUlM0YlMjYqKCk6ISU0MCUyMyUyNCUyNSUzRCU1RSUyNiolM0QoKV8lN0IlN0QlM0ElM0UlM0YlM0UlM0M=" has a length of 130.

Fails because they don't decode the same (if you can believe that).  I found an article last week that explained the differences in the various implementations, but that lead me to this solution as well.

Reply:
that's odd. Thanks for trying